nautobot/docs/plugins/development.md

```diff
@@ -574,6 +574,136 @@ A `PluginMenuButton` has the following attributes:
 !!! note
     Any buttons associated within a menu item will be shown only if the user has permission to view the link, regardless of what permissions are set on the buttons.
 
+## Navigation Menu Tabs
+
+Plugins can modify the existing navigation bar layout by using `nav_tabs` inside of `navigation.py`. Using the key and weight system, a developer can integrate the plugin amongst existing menu tabs, groups, items and buttons and/or create entirely new menus as desired.
+
+### Adding a new tab
```

Comment:
Thinking about it, this whole section of documentation is now really applicable to both plugins and core apps, isn't it? I think it might be more appropriate to create a new document under `docs/development/` and link to it from the plugin docs as needed.